### PR TITLE
Fix translations loading for Jetpack plans

### PR DESCRIPTION
**Pull request by dlind1**
#### Changes proposed in this Pull Request

Refactor the retrieval of plans array to account for situations where the translations for all the plans become available later than the translation for the current plan name. Originally addressed in https://github.com/Automattic/wp-calypso/pull/52629

#### Testing instructions

* Switch to a non-English user locale - non-Latin character set would show the issue more clearly.
* Have a linked Jetpack site, navigate to https://wordpress.com/plans/your-jetpack-site-url, reload the browser tab a few times until 'Security Daily', 'Complete' and 'Security Real-Time' display English titles and descriptions.
* Checkout the branch locally and start with `BUILD_TRANSLATION_CHUNKS=true ENABLE_FEATURES=use-translation-chunks,wpcom-user-bootstrap yarn start`
* Navigate to http://calypso.localhost:3000/plans/your-jetpack-site-url and confirm that all the plans are being translated now, even after multiple refreshes.

Related to #52629

### CODE REVIEW

File: client/my-sites/plans/jetpack-plans/product-grid/index.tsx [hunk 1]
```diff
@@ -61,24 +61,38 @@ const ProductGrid: React.FC< ProductsGridProps > = ( {
...
+		[ duration, currentPlanSlug ]
+	);
+	// Get the first plan description and pass it through `translate` so that
+	// if it wasn't translated at the start the cache key for sortedPlans
+	// would change once translation becomes available, but if it was translated
+	// from the start the call to `translate` won't have any effect on it.
+	const translatedFirstPlanDescription =
+		untranslatedSortedPlans &&
+		untranslatedSortedPlans[ 0 ] &&
+		untranslatedSortedPlans[ 0 ].description
```
Comment (yuliyan) [inline client/my-sites/plans/jetpack-plans/product-grid/index.tsx]: Can be simplified with optional chaining - `untranslatedSortedPlans?.[ 0 ]?.description`.

**Review by yuliyan — COMMENTED** [review]
Tested locally, but I'm still getting the untranslated features strings:
image

I think that comes from the fact that that plan descriptions and features strings are in a separate chunks. So the plan description translation, that you use as a flag, is being loaded separately from the features translations and therefore the plans array is not being recomputed when the latter translations get loaded. 

Note that I had to bypass #52791 to be able to reproduce the problem

File: client/my-sites/plans/jetpack-plans/product-grid/index.tsx (hunk 1 above)
Comment (dlind1) [reply client/my-sites/plans/jetpack-plans/product-grid/index.tsx]: Done

**Review by yuliyan — APPROVED** [review]
Tested, LGTM!

**Pull request merged**